src/wasm/wasm-binary.cpp

```diff
@@ -2280,8 +2339,65 @@ void WasmBinaryBuilder::readNames(size_t payloadLen) {
           }
         }
       }
+    } else if (nameType == BinaryConsts::UserSections::Subsection::NameTable) {
+      auto num = getU32LEB();
+      for (size_t i = 0; i < num; i++) {
+        auto index = getU32LEB();
+        auto rawName = getInlineString();
+        rawName = escape(rawName);
+        auto name = rawName;
+        if (index == 0) {
+          wasm.table.name = name;
+        } else {
+          std::cerr << "warning: table index out of bounds in name section, "
+                       "table subsection: "
+                    << std::string(name.str) << " at index "
+                    << std::to_string(index) << std::endl;
+        }
+      }
+    } else if (nameType == BinaryConsts::UserSections::Subsection::NameMemory) {
+      auto num = getU32LEB();
+      for (size_t i = 0; i < num; i++) {
+        auto index = getU32LEB();
+        auto rawName = getInlineString();
+        rawName = escape(rawName);
+        auto name = rawName;
```

Comment:
Is this supposed to look like this?

```suggestion
        auto name = escape(rawName);
```

Could also do `auto name = escape(getInlineString());`.

Reply:
This is an exact copy & paste from what was already done when reading functions

https://github.com/WebAssembly/binaryen/blob/9de15862c1bc4e2b092ccbe89fdb4b850704bbfc/src/wasm/wasm-binary.cpp#L2228-L2230

Going to update it there as well :)